next-level/lib/dependencies.py

```
@@ -0,0 +1,273 @@
+
+def check_binary(name: str) -> bool:
+    """Check if a binary is available on PATH."""
+    return shutil.which(name) is not None
+
+
+def check_binary_version(name: str) -> str | None:
+    """Get version string of a binary, or None if not available."""
+    if not check_binary(name):
+        return None
+    try:
+        result = subprocess.run(
```

Ruff (0.15.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)